satellite/satellitedb/overlaycache_test.go

```diff
@@ -0,0 +1,180 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb_test
+
+import (
+	"github.com/lib/pq"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+	"storj.io/storj/satellite/satellitedb/satellitedbtest"
+)
+
+func TestOverlaycache_AllPieceCounts(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	satellitedbtest.Run(t, func(t *testing.T, db satellite.DB) {
+		// get overlay db
+		overlay := db.OverlayCache()
+
+		// get dbx db access
+		_db, ok := db.(interface{ TestDBAccess() *dbx.DB })
+		require.True(t, ok)
+
+		dbAccess := _db.TestDBAccess()
+		require.NotNil(t, dbAccess)
+
+		// create test nodes in overlay db
+		testNodes := createTestNodes(ctx, 10, t, dbAccess)
+
+		// set expected piece counts
+		var err error
+		expectedPieceCounts := make(map[storj.NodeID]int)
+		for i, node := range testNodes {
+			pieceCount := math.Pow10(i + 1)
+			nodeID, err := storj.NodeIDFromBytes(node.Id)
+			require.NoError(t, err)
+
+			expectedPieceCounts[nodeID] = int(pieceCount)
+		}
+
+		updateTestPieceCounts(t, dbAccess, expectedPieceCounts)
+		require.NoError(t, err)
+
+		// expected and actual piece count maps should match
+		actualPieceCounts, err := overlay.AllPieceCounts(ctx)
+		require.NoError(t, err)
+		require.Equal(t, expectedPieceCounts, actualPieceCounts)
+	})
+}
+
+func TestOverlaycache_UpdatePieceCounts(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	satellitedbtest.Run(t, func(t *testing.T, db satellite.DB) {
+		// get dbx db access
+		_db, ok := db.(interface{ TestDBAccess() *dbx.DB })
+		require.True(t, ok)
+
+		dbAccess := _db.TestDBAccess()
+		require.NotNil(t, dbAccess)
+
+		// create test nodes in overlay db
+		testNodes := createTestNodes(ctx, 10, t, dbAccess)
+
+		// set expected piece counts
+		expectedPieceCounts := testPieceCounts(t, testNodes)
+
+		// update piece count fields on test nodes; set exponentially
+		updateTestPieceCounts(t, dbAccess, expectedPieceCounts)
```

Comment:
You should be calling `overlayService.UpdatePieceCounts` in this test instead of `updateTestPieceCounts` I think